pkg/asset/installconfig/platformcredscheck.go

```diff
@@ -86,3 +102,64 @@ func (a *PlatformCredsCheck) Generate(dependencies asset.Parents) error {
 func (a *PlatformCredsCheck) Name() string {
 	return "Platform Credentials Check"
 }
+
+// fileExists reports whether the file exists.
+func fileExists(path string) (bool, error) {
+	info, err := os.Stat(path)
+	if err == nil && !info.IsDir() {
+		return true, nil
+	}
+	if err != nil && os.IsNotExist(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+// determineOpenStackConfigLocation sequentially searches through the
+// possible places where clouds.yaml can be located and returns the first
+// location found.
+func determineOpenStackConfigLocation() string {
+	// Currently the search order is:
+	// 1. OS_CLIENT_CONFIG_FILE env variable
+	// 2. Current directory.
+	// 3. unix-specific user_config_dir (~/.config/openstack/clouds.yaml)
+	// 4. unix-specific site_config_dir (/etc/openstack/clouds.yaml)
+	//
+	// For more information:
+	// https://github.com/gophercloud/utils/blob/master/openstack/clientconfig/utils.go#L95-L98
+
+	yamlFile := "clouds.yaml"
+
+	// OS_CLIENT_CONFIG_FILE
+	if path := os.Getenv("OS_CLIENT_CONFIG_FILE"); path != "" {
+		if ok, _ := fileExists(path); ok {
+			return path
+		}
+	}
+
+	// Current directory.
+	cwd, _ := os.Getwd()
+	path := filepath.Join(cwd, yamlFile)
+	if ok, _ := fileExists(path); ok {
+		return path
+	}
+
+	// unix user config directory: ~/.config/openstack.
+	if currentUser, err := user.Current(); err == nil {
+		homeDir := currentUser.HomeDir
+		if homeDir != "" {
+			path := filepath.Join(homeDir, ".config/openstack/"+yamlFile)
```

Comment:
I was about complain that it ignores `$XDG_CONFIG_HOME`, but this comes directly from gophercloud:

https://github.com/gophercloud/utils/blob/master/openstack/clientconfig/utils.go#L95-L98

*sigh*